statsd/reporter.go

```diff
@@ -0,0 +1,63 @@
+package statsd
+
+import (
+	"sync"
+	"time"
+
+	"github.com/cactus/go-statsd-client/statsd"
+	"github.com/uber-go/tally"
+)
+
+type cactusStatsReporter struct {
+	statter  statsd.Statter
+	sm       sync.Mutex
+	scopes   []tally.Scope
+	interval time.Duration
+	quit     chan struct{}
+}
+
+// NewStatsdReporter wraps a statsd.Statter for use with tally. Use either statsd.NewClient or statsd.NewBufferedClient.
+func NewStatsdReporter(statsd statsd.Statter, interval time.Duration) tally.StatsReporter {
+	return &cactusStatsReporter{
+		quit:     make(chan struct{}),
+		statter:  statsd,
+		interval: interval,
+		scopes:   make([]tally.Scope, 0),
+	}
+}
+
+func (r *cactusStatsReporter) Start() {
+	ticker := time.NewTicker(r.interval)
+	for {
+		select {
+		case <-ticker.C:
+			for _, scope := range r.scopes {
+				scope.Report(r)
+			}
+		case <-r.quit:
+			return
+		}
+	}
+}
+
+func (r *cactusStatsReporter) registerScope(s tally.Scope) {
```

Comment:
Like it says on the tin. A single reporter can have multiple scopes. E.g. creating a SubScope. This method is required to update the Reporter's internal bookkeeping.